znoyder/generator.py

```diff
@@ -155,12 +155,14 @@ def list_existing_osp_projects() -> list:
 def cleanup_generated_jobs_dir() -> None:
     if os.path.exists(GENERATED_CONFIGS_DIR):
         rmtree(GENERATED_CONFIGS_DIR)
-        LOG.info('Removed the directory: {GENERATED_CONFIGS_DIR}')
+        LOG.info(f'Removed the directory: {GENERATED_CONFIGS_DIR}')
     destination_directory = os.path.dirname(GENERATED_CONFIGS_DIR)
     Path(destination_directory).mkdir(parents=True, exist_ok=True)
 
 
 def main(args) -> None:
+
+    logger.ensure_correct_logger(args)
```

Comment:
Can we make it a global parameter that would be set for all subcommands?

Reply:
Just to clarify: you mean to give the same option to all subcommands, or to make them a general option for znoyder? Both are easily doable, but I feel like the first option would be the most user friendly.

Reply:
I was thinking about a general option for Znoyder, as we use the same logger in most of the components already (if not all right now, we will in the future), so IMO that's the better solution.
Why do you think having such option for each component would be more user friendly? I am afraid that would be mostly a redundancy, but maybe I miss something.

Reply:
Maybe user friendly is not really the right wording. I actually tried this after your first comment and what I did not like is that the logging options should be specified before the command, that is
`znoyder --log-file output.txt generate --component ...`
which I feel might be confusing and users might forget to put those options before and get errors.

Reply:
Ach, right, I remember something like this, it's some limitation of argparser with subparsers. This is definitely a user-friendly argument. Though I recall I saw some workaround for this, but I am not sure right now, but I think with small refactor of argparsers it should be achievable.

Ok, so I am fine with keeping it how it is and we may propose a dedicated task to research this workaround (or you can do it as part of this task, if you have some sprare working hours :smiley:)

Reply:
If I may chime in, I think the most user-friendly option is to not let the user think where he should put these - so basically support it before the sub command and after.